# Importing Package and sourcing data

In [31]:
import os
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import seaborn as sns
import warnings
pd.options.mode.chained_assignment = None  
warnings.filterwarnings('ignore')


In [32]:
df = pd.read_csv("kalimati_tarkari_dataset_cleaned.csv")

# Analysing data source

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 197161 entries, 0 to 197160
Data columns (total 7 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   SN         197161 non-null  int64  
 1   Commodity  197161 non-null  object 
 2   Date       197161 non-null  object 
 3   Unit       197161 non-null  object 
 4   Minimum    197161 non-null  float64
 5   Maximum    197161 non-null  float64
 6   Average    197161 non-null  float64
dtypes: float64(3), int64(1), object(3)
memory usage: 10.5+ MB


There are 197161 data; out of which all of them are not-null type, we can say that the data is of high quality

In [34]:
#parsing the date as date
df['Date'] = pd.to_datetime(df['Date'])

In [35]:
#Looking at few top data 
df.head()

SN            Commodity       Date Unit  Minimum  Maximum  Average
0   0   Tomato Big(Nepali) 2013-06-16   Kg     35.0     40.0     37.5
1   1  Tomato Small(Local) 2013-06-16   Kg     26.0     32.0     29.0
2   2           Potato Red 2013-06-16   Kg     20.0     21.0     20.5
3   3         Potato White 2013-06-16   Kg     15.0     16.0     15.5
4   4   Onion Dry (Indian) 2013-06-16   Kg     28.0     30.0     29.0

In [36]:
df['Commodity'].nunique()

132

As we can see that the data consist of 132 different commodities

In [37]:
#Lets see the range of the date available
date_range = str(df['Date'].min()) + ' - ' + str(df['Date'].max())
date_range_days = (df['Date'].max() - df['Date'].min()).days

print("Consists of data from", date_range, "in total days:", date_range_days)


Consists of data from 2013-06-16 00:00:00 - 2021-05-13 00:00:00 in total days: 2888


Consists of data from 2013-06-16 00:00:00 - 2021-05-13 00:00:00 in total days: 2888
